File: test_api.js
const fetch = require('node-fetch');

async function test() {
    try {
        const res = await fetch("http://127.0.0.1:8000/api/stats/advanced_player_metrics?year=2025");
        const data = await res.json();
        const offense = data.filter(p => ['QB','RB','WR','TE'].includes(p.position));
        console.log(`Offense players: ${offense.length}`);
        if(offense.length > 0) {
            console.log(offense[0]);
        }
    } catch(e) {
        console.error(e);
    }
}
test();

File: test_api.py
import urllib.request, json
try:
    resp = urllib.request.urlopen("http://localhost:8000/api/stats/advanced_player_metrics?year=2025")
    data = json.loads(resp.read())
    offense = [p for p in data if p.get('position') in ['QB', 'RB', 'WR', 'TE']]
    with open('offense_test.json', 'w') as f:
        json.dump(offense[:3], f, indent=2)
except Exception as e:
    with open('offense_test.json', 'w') as f:
        f.write(str(e))
